src/middleware/authMiddleware.ts

```diff
@@ -5,33 +5,114 @@ import { JwtPayload } from "../types/auth";
 import { User } from "../models/User";
 import { AppDataSource } from "../config/db";
 
-declare global {
-  namespace Express {
-    interface Request {
-      user?: JwtPayload;
-    }
-  }
-}
-
-export const authenticateToken = (
+export const authenticateToken = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
-  const authHeader = req.headers.authorization;
-  const token = authHeader && authHeader.split(" ")[1];
+
+  const authHeader = 
+    req.headers['authorization'] || 
+    req.headers['Authorization'] || 
+    req.headers['AUTHORIZATION'];
```

Comment:
This is technically correct but semantically incorrect, because HTTP headers are case-insensitive and will automatically the different caps.
```suggestion
    const authHeader = req.headers['authorization'];

```

Reply:
Alright, noted...